src/main/webapp/js/fetch_handler.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Formats the errors from a fetch made to the Open Sesame API.
+ * There are three different types of errors that are formatted here: a network
+ * error from the Fetch API, a server error from the Open Sesame API, and a
+ * defined API error from the Open Sesame API.
+ *
+ * All error responses from the Open Sesame API should contain a JSON body
+ * formatted to match the ErrorResponse object here. {@link ErrorResponse}.
+ */
+/**
+ * @typedef ErrorResponse
+ * @property {string} error
+ * @property {number} [statusCode]
```

Comment:
Just a Q: What does it mean that statusCode is in brackets?

Reply:
It indicates that the `statusCode` property is optional and will potentially not be present in some `ErrorResponse` objects. Good eye!

You can read about it [here](https://jsdoc.app/tags-type.html) (Ctrl+F "optional" to find it).